rust/user-info-fetcher/src/http_error.rs

```diff
@@ -0,0 +1,55 @@
+use std::sync::Arc;
+
+use axum::{response::IntoResponse, Json};
+use hyper::StatusCode;
+use serde::Serialize;
+
+pub trait Error: std::error::Error {
+    fn status_code(&self) -> StatusCode;
+}
+impl<T: Error> Error for Arc<T> {
+    fn status_code(&self) -> StatusCode {
+        let inner: &T = self;
+        inner.status_code()
+    }
+}
+
+pub struct JsonResponse<E> {
```

Comment:
I feel like `JsonResponse` might not be the best name for how the struct is used (indicating an error occurred). How about `JsonErrorResponse` or something similar?

Reply:
And remove the generic? Yeah could do. This should all move to operator-rs anyway.

Reply:
Oh actually the generic E looks like it needs to stay based on the other lines below

Reply:
The intention here is that you'd use it as `error::JsonResponse`. And yeah the generic is inherent to both the data storage and the serde impls.

Reply:
I don't have any super strong opinions on that, so we can move forward with it for now.